### PR TITLE
README improvements, minor cleanups

### PR DESCRIPTION
**Pull request by StephanTLavavej**
* Consistently follow our convention of defining variables on separate lines.
* Simplify `FBITS` logic - it's always `24` (single-precision) or `53` (double-precision) for us.
  + This is now consistent with: https://github.com/microsoft/STL/blob/484fbc9742f895030047977efc55dc8950a37086/stl/src/xxxdtent.hpp#L18-L42
* Wrap a slightly-too-long line in `provision-image.ps1`.
  + I verified that the script still succeeds.
* **Improve the README** and avoid long lines.
  + Add links to our Status Chart and Discord at the very top, where they'll be noticed.
  + Organize them in a bulleted list below the Changelog, for even more prominence.
  + Slightly rephrase the IDE instructions to make them easier to scan (and avoid a long line). Drop "to install the boost-math dependency" as this is apparent from the command and the previous explanation.
  + Moderately reorganize the command prompt instructions. (I verified that they can be followed from scratch.)
    - Talking about changing the `x64-windows` constant was confusing - we're already depicting boost-math being installed for both `x86-windows` and `x64-windows`. What we really mean is to open up a different kind of Native Tools Command Prompt, and adjust the build directory accordingly. These instructions are so short, we can just depict them directly, avoiding all "you know what we mean" ambiguity.
    - The initial "one time" steps of cloning the repo and building vcpkg don't need a VS command prompt; rephrase the step to "Open a command prompt." to indicate that it doesn't matter.
    - The command-line instructions don't need to say "Invoke". Just mention the command.
    - Again, "to install the boost-math dependency" wasn't providing useful information; strike it.
    - Drop the "wherever you want binaries" generalization. We specifically `.gitignore` the `/out/` directory. Our instructions are simple enough for advanced users to understand what paths they can replace.
    - (Longer-term, I would like to restructure the README even more along these lines, offering default incantations to build, update the paths, and run the tests - this is a step in that direction.)
  + Depict `C:\Users\username\Desktop` to follow our usual policies. (I would actually prefer `C:\Temp` here, but users tend to think that that is somehow special. Maybe `C:\Anywhere` instead?)
  + Adjust the testing command line examples to stay within 120 columns, by extracting the current directory.
    - Note that there's a slight adjustment here, as we're increasingly harmonizing our depicted paths - change `C:\STL` to `C:\Dev\STL` which was used in previous examples.
* Validate line length for `.cmd`, `.md`, `.ps1`, `.py`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -147,30 +150,39 @@ acquire this dependency.
...
+To build the x86 target:
+
+1. Open an "x86 Native Tools Command Prompt for VS 2019".
+2. Change directories to the previously cloned `STL` directory.
+3. `cmake -G Ninja -S . -B out\build\x86`
+4. `ninja -C out\build\x86`
+
+To build the x64 target:
+
+1. Open an "x64 Native Tools Command Prompt for VS 2019".
```
Comment (cbezault) [inline README.md:182]: This is sort of not totally true, the user wants an x64 targeting compiler right? Not explicitly the x64 native compiler. I guess it's ok to assume everyone's on x64 these days though so no changes requested.

Comment (StephanTLavavej) [reply README.md:182]: Note that this doesn't claim to the be the only possible way, just the recommended way.

(Also, the VS-installed clang-cl works in native command prompts only, not cross, so the difference is relevant for later testing.)

File: README.md [hunk 2]
```diff
@@ -244,29 +256,20 @@ under a category in libcxx, or running a single test in `std` and `tr1`.
...
-```
+These examples assume that your current directory is `C:\Dev\STL\out\build\x64`.
+
+* This command will run all of the testsuites with verbose output.
+  + `ctest -V`
+* This command will also run all of the testsuites.
+  + `python tests\utils\stl-lit\stl-lit.py ..\..\..\llvm-project\libcxx\test ..\..\..\tests\std ..\..\..\tests\tr1`
+* This command will run all of the std testsuite.
+  + `python tests\utils\stl-lit\stl-lit.py ..\..\..\tests\std`
+* If you want to run a subset of a testsuite you need to point it to the right place in the sources. The following
```
Comment (mnatsuhara) [inline README.md:267]: For some reason, my brain wants a comma here.  Feel free to ignore, as I'm not entirely sure it's grammatically required or even appropriate.  
```suggestion
* If you want to run a subset of a testsuite, you need to point it to the right place in the sources. The following
```

File: README.md [hunk 3]
```diff
@@ -244,29 +256,20 @@ under a category in libcxx, or running a single test in `std` and `tr1`.
...
+will run the single test found under VSO_0000000_any_calling_conventions.
+  + `python tests\utils\stl-lit\stl-lit.py ..\..\..\tests\std\tests\VSO_0000000_any_calling_conventions`
+* You can invoke stl-lit with any arbitrary subdirectory of a testsuite. In libcxx this allows you to have finer
```
Comment (mnatsuhara) [inline README.md:270]: It's clear enough without this change, but seems like it should be consistent with our general `codify` all the command-y things. 
```suggestion
* You can invoke `stl-lit` with any arbitrary subdirectory of a testsuite. In libcxx this allows you to have finer
```

**Comment by StephanTLavavej** [comment]
@mnatsuhara Thanks - I've applied your suggestions!

**Pull request merged**